File: praktikum24/app/Models/Prestasi.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Prestasi extends Model
{
    protected $table = 'prestasi';
    protected $fillable = [
        'mahasiswa_id', 'jenis_id', 'tingkat_id',
        'nama_lomba', 'penyelenggara', 'tanggal',
        'juara', 'sertifikat', 'status_verifikasi',
    ];

    protected $casts = ['tanggal' => 'date'];

    public function mahasiswa()
    {
        return $this->belongsTo(Mahasiswa::class, 'mahasiswa_id');
    }

    public function jenis()
    {
        return $this->belongsTo(JenisPrestasi::class, 'jenis_id');
    }

    public function tingkat()
    {
        return $this->belongsTo(TingkatPrestasi::class, 'tingkat_id');
    }

    public function verifikasi()
    {
        return $this->hasOne(VerifikasiPrestasi::class, 'prestasi_id');
    }
}

File: simak_mobile/lib/data/models/prestasi.dart
import 'jenis_tingkat_prestasi.dart';
import 'mahasiswa.dart';

class Prestasi {
  final int id;
  final Mahasiswa? mahasiswa;
  final JenisPrestasi? jenis;
  final TingkatPrestasi? tingkat;
  final String? namaLomba;
  final String? penyelenggara;
  final String? tanggal;
  final String? juara;
  final String? sertifikat;
  final String? statusVerifikasi;

  Prestasi({
    required this.id,
    this.mahasiswa,
    this.jenis,
    this.tingkat,
    this.namaLomba,
    this.penyelenggara,
    this.tanggal,
    this.juara,
    this.sertifikat,
    this.statusVerifikasi,
  });

  factory Prestasi.fromJson(Map<String, dynamic> json) {
    return Prestasi(
      id: json['id'],
      mahasiswa: json['mahasiswa'] != null
          ? Mahasiswa.fromJson(json['mahasiswa'] as Map<String, dynamic>)
          : null,
      jenis: json['jenis'] != null
          ? JenisPrestasi.fromJson(json['jenis'] as Map<String, dynamic>)
          : null,
      tingkat: json['tingkat'] != null
          ? TingkatPrestasi.fromJson(json['tingkat'] as Map<String, dynamic>)
          : null,
      namaLomba: json['nama_lomba'],
      penyelenggara: json['penyelenggara'],
      tanggal: json['tanggal'],
      juara: json['juara'],
      sertifikat: json['sertifikat'],
      statusVerifikasi: json['status_verifikasi'],
    );
  }
}
